server/filestore.go

```diff
@@ -4794,14 +4801,20 @@ func (fs *fileStore) removeMsg(seq uint64, secure, viaLimits, needFSLock bool) (
 	// If we are tracking subjects here make sure we update that accounting.
 	mb.ensurePerSubjectInfoLoaded()
 
+	// ensurePerSubjectInfoLoaded can recreate the cache, so we need to recapture it
+	// before writing to it, otherwise we can lose the entries.
+	if mbcache = mb.cache.Value(); mbcache == nil {
+		goto retry
```

Comment:
We can go quite far back here when doing `goto retry`. Wouldn't the `mb.msgs` and `mb.bytes` accounting be duplicated in that case?
Should there be a similar `retry2:` block here?